icons/src/fullcolor/render-bitmaps-verbose.py

```diff
@@ -238,60 +221,278 @@ def characters(self, chars):
+    print("│   path of Inkscape executable                    │")
+    print("│   (if the script can't find it)                  │")
+    print("│                                                  │")
+    print("│ --individual_icons [INDIVIDUAL_ICONS]            │")
+    print("│                                                  │")
+    print("│   individual icon names (without extensions)     │")
+    print("│   to render (by default all)                     │")
+    print("│                                                  │")
+    print("│ --optipng_path OPTIPNG_PATH                      │")
+    print("│                                                  │")
+    print("│   path of OptiPNG executable                     │")
+    print("│   (if the script can't find it)                  │")
+    print("│                                                  │")
+    print("│ --output_path OUTPUT_PATH                        │")
+    print("│                                                  │")
+    print("│   output directory (by default '../../Suru')     │")
+    print("│                                                  │")
+    print("│ --scaling_factors [SCALING_FACTORS]              │")
+    print("│                                                  │")
+    print("│   scaling factors to render at                   │")
+    print("│   (by default [1, 2], e.g. 100% & 200%)          │")
+    print("│                                                  │")
+    print("│ --verbose                                        │")
+    print("│                                                  │")
+    print("│   print verbose output to the terminal           │")
+    print("│                                                  │")
+    print("└──────────────────────────────────────────────────┘")
+    print("")
+    return
+
+
+def print_categories():
 
-if args.svg is None:
-    print("\nNo arguments provided; processing listed sources:\n")
-    for source in CATEGORIES_TO_RENDER:
-        print("-- \"" + os.path.join('.', source) + "\"")
     print("")
-else:
     print("┌────────────────────────────────┐")
-    print("│ Rendering from command-line argument \"" + args.svg + "\"")
+    print("│ Icon Categories:               │")
     print("├────────────────────────────────┤")
+    print("│ - actions                      │")
+    print("│ - apps                         │")
+    print("│ - categories                   │")
+    print("│ - devices                      │")
+    print("│ - emblems                      │")
+    print("│ - legacy                       │")
+    print("│ - mimetypes                    │")
+    print("│ - places                       │")
+    print("│ - status                       │")
+    print("│ - wip                          │")
+    print("└────────────────────────────────┘")
+    print("")
+    return
+
+
+def build_parser():
+
+    parser = argparse.ArgumentParser(add_help=False)
```

Comment:
I love the effort and the help output is wonderful, however it seems a bit maintenance unfriendly :) Any change in the arguments will require to update the help message too. Was the default help message so bad?

Reply:
The default help message was kind of a mess specifically because it doesn’t like longer argument names. I could presumably make the formatting of the custom help message more programmatic, but it might take a little bit.